charts/helm-connector-backend/connector-backend/templates/deployment.yaml

```diff
@@ -0,0 +1,240 @@
+            - name: volume-tls-secret-ca
+              mountPath: "/etc/ssl/certs/ca.crt"
+              subPath: "ca.crt"
+              readOnly: true
+            - name: volume-tls-secret
+              mountPath: "/ssl"
+              readOnly: true
+            {{- end }}
+            {{- if .Values.deployment.extraVolumeMounts }}
+            {{- toYaml .Values.deployment.extraVolumeMounts | indent 12 }}
+            {{- end }}
+          env:
+            - name: CFG_DATABASE_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.deployment.config.database.secret.name }}"
+                  key: "{{ .Values.deployment.config.database.secret.usernamekey }}"
+            - name: CFG_DATABASE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.deployment.config.database.secret.name }}"
+                  key: "{{ .Values.deployment.config.database.secret.passwordkey }}"
+            - name: CFG_DATABASE_HOST
+              value: "{{ .Values.deployment.config.database.host }}"
+            - name: CFG_DATABASE_PORT
+              value: "{{ .Values.deployment.config.database.port }}"
+            - name: CFG_SERVER_DISABLEUSAGE
+              value: "{{ .Values.deployment.config.server.disableusage }}"
+            - name: CFG_SERVER_EDITION
+              value: "{{ .Values.deployment.config.server.edition }}"
+            - name: CFG_MGMTBACKEND_HOST
+              value: "{{ .Values.deployment.config.mgmtbackend.host }}"
+            - name: CFG_MGMTBACKEND_PORT
+              value: "{{ .Values.deployment.config.mgmtbackend.port }}"
+            - name: CFG_PIPELINEBACKEND_HOST
+              value: "{{ .Values.deployment.config.pipelinebackend.host }}"
+            - name: CFG_PIPELINEBACKEND_PORT
+              value: "{{ .Values.deployment.config.pipelinebackend.port }}"
+            - name: CFG_TEMPORAL_CLIENTOPTIONS_HOSTPORT
+              value: "{{ .Values.deployment.config.temporal.clientoptions.hostport }}"
+            {{- if and .Values.deployment.config.server.https.cert .Values.deployment.config.server.https.key }}
+            - name: CFG_SERVER_HTTPS_CERT
+              value: "{{ .Values.deployment.config.server.https.cert }}"
+            - name: CFG_SERVER_HTTPS_KEY
+              value: "{{ .Values.deployment.config.server.https.key }}"
+            - name: CFG_MGMTBACKEND_HTTPS_CERT
+              value: "{{ .Values.deployment.config.mgmtbackend.https.cert }}"
+            - name: CFG_MGMTBACKEND_HTTPS_KEY
+              value: "{{ .Values.deployment.config.mgmtbackend.https.key }}"
+            - name: CFG_PIPELINEBACKEND_HTTPS_CERT
+              value: "{{ .Values.deployment.config.pipelinebackend.https.cert }}"
+            - name: CFG_PIPELINEBACKEND_HTTPS_KEY
+              value: "{{ .Values.deployment.config.pipelinebackend.https.key }}"
```

Comment:
All configuration variables should be set by ConfigMap and mounted into container to overwrite the default `config/config.yaml`.